sound/soc/sof/loader.c

```diff
@@ -66,6 +66,8 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 			ret = get_ext_windows(sdev, ext_hdr);
 			break;
 		default:
+			dev_warn(sdev->dev, "warning: unknown ext header type %d size 0x%x\n",
+				 ext_hdr->type, ext_hdr->hdr.size);
```

Comment:
@ktrzcinx wondering if it makes sense to keep on going if we ever hit a unknown ext header?

Reply:
It's a half-full question. You could
a) consider it's harmless and move to the next chunk, which enables an old kernel to deal with a new firmware and just skip information it doesn't understand.
b) be paranoid and consider that if the type is incorrect then the odds of the size being wrong are very high. if you look at the code in get_ext_windows we don't accept unknown types

In this case, the patch only adds a warning and the existing flow is not changed, so I am tempted to go with a)

Reply:
> the odds of the size being wrong are very high
I don't think so, in each header full packet size must be specified, so even if kernel don't understand new frame then it can be securely skipped - it is why it is named extended header. When size is wrong then there is an issue with frame itself, not with parser. This is why I choose option a)

Reply:
firmware files can be cooked to break the kernel, it's a well-known attack surface. yes in the nominal case this cannot happen with regular firmware, but nothing prevents a malicious change to a firmware file...